api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -66,6 +66,10 @@ type ClientTrafficPolicySpec struct {
 	//
 	// +optional
 	EnableProxyProtocol *bool `json:"enableProxyProtocol,omitempty"`
+	// ClientIPDetectionSettings provides XFF and extension configuration for client IP detection on the listener.
+	//
+	// +optional
+	ClientIPDetection *ClientIPDetectionSettings `json:"clientIPDetection,omitempty"`
```

Comment:
```suggestion
	ClientIPDetection *ClientIPDetectionSettings `json:"clientIpDetection,omitempty"`
```

ptal @envoyproxy/gateway-maintainers